pangeo_forge_recipes/recipes/xarray_zarr.py

```diff
@@ -549,20 +571,20 @@ def _gather_coordinate_dimensions(group: zarr.Group) -> List[str]:
     )
 
 
-def finalize_target(*, config: XarrayZarrRecipe) -> None:
-    if config.target is None:
-        raise ValueError("target has not been set.")
-
-    if config.consolidate_dimension_coordinates:
-        logger.info("Consolidating dimension coordinate arrays")
-        target_mapper = config.target.get_mapper()
-        group = zarr.open(target_mapper, mode="a")
-        # https://github.com/pangeo-forge/pangeo-forge-recipes/issues/214
-        # intersect the dims from the array metadata with the Zarr group
-        # to handle coordinateless dimensions.
+def _consolidate_dimension_coordinate(
```

Comment:
The diff here looks strange.

All I did was

1. move the `if config.consolidate_dimension_coordinates` to its own function so that I could call it from two places (here and earlier in `prepare_target` for `concat_dim`)
2. Let the caller pass in the `dims` to check (so that we can do just `concat_dim` early on, and the other dims later; I think this is what we want).
3. Added a check that `arr.chunks != arr.shape (to avoid rewriting `concat_dim` in `finalize_target`, since it would already be consolidated)